rotkehlchen/tests/exchanges/test_woo.py

```diff
@@ -0,0 +1,449 @@
+import warnings as test_warnings
+from json.decoder import JSONDecodeError
+from unittest.mock import MagicMock, call, patch
+
+import requests
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.converters import asset_from_woo
+from rotkehlchen.constants.assets import A_BTC, A_ETH, A_SOL, A_USDT, A_WOO
+from rotkehlchen.errors.asset import UnknownAsset
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.exchanges.data_structures import (
+    AssetMovement,
+    AssetMovementCategory,
+    Trade,
+    TradeType,
+)
+from rotkehlchen.exchanges.woo import API_MAX_LIMIT, Woo
+from rotkehlchen.fval import FVal
+from rotkehlchen.types import Location, Timestamp
+from rotkehlchen.utils.misc import ts_sec_to_ms
+
+
+def test_name():
+    exchange = Woo('woo', 'a', b'a', object(), object())
+    assert exchange.location == Location.WOO
+    assert exchange.name == 'woo'
+
+
+def test_woo_assets_are_known(mock_woo):
+    request_url = f'{mock_woo.base_uri}/v1/public/token'
+    try:
+        response = requests.get(request_url)
+    except requests.exceptions.RequestException as e:
+        raise RemoteError(
+            f'Woo get request at {request_url} connection error: {e!s}.',
+        ) from e
+    if response.status_code != 200:
+        raise RemoteError(
+            f'Woo query responded with error status code: {response.status_code} '
+            f'and text: {response.text}',
+        )
+    try:
+        response_list = response.json()
+    except JSONDecodeError as e:
+        raise RemoteError(f'Woo returned invalid JSON response: {response.text}') from e
+    for row in response_list['rows']:
+        try:
+            asset_from_woo(row['token'])
+        except UnknownAsset as e:
+            test_warnings.warn(UserWarning(
+                f'Found unknown asset {e.identifier} in {mock_woo.name}. '
+                f'Support for it has to be added',
+            ))
+
+
+def test_query_online_trade_history_basic(mock_woo):
+    """Assert that the expected arguments are passed to the `_api_query` method"""
+    start_ts, end_ts = (1634600000, 1634620000)
```

Comment:
no need for parentheses

```suggestion
    start_ts, end_ts = 1634600000, 1634620000
```